providers/flagd/src/main/java/dev/openfeature/contrib/providers/flagd/FlagdProvider.java

```diff
@@ -183,5 +190,12 @@ private void handleStateTransition(ProviderState oldState, ProviderState newStat
             this.emitProviderReady(details);
             this.emitProviderConfigurationChanged(details);
         }
+        // we shutdown from an error
+        if (ProviderState.ERROR.equals(oldState) && ProviderState.NOT_READY.equals(newState)) {
```

Comment:
This condition will not be reached as there's `if (ProviderState.NOT_READY.equals(newState))` in line 167

Reply:
Besides, this check is not needed. The state can transit from Error to Shutdown [1]

[1] - https://openfeature.dev/specification/sections/flag-evaluation#17-provider-lifecycle-management

Reply:
Fixed.